### PR TITLE
Implement fixture scenarios and refactor module loading

### PR DESCRIPTION
**Pull request by g-cassie**
Re https://github.com/samselikoff/ember-cli-mirage/issues/136#issuecomment-105636546

Need feedback on whether/how to test the new scenarios functionality. Hope that my refactor of the module loading makes sense.  Open to other ideas on how to structure it.

### CODE REVIEW

File: app/initializers/ember-cli-mirage.js [hunk 1]
```diff
@@ -22,11 +23,13 @@ export default {
         server.loadConfig(testConfig);
       }
 
-      if (env === 'test' && factoryMap) {
-        server.loadFactories(factoryMap);
-
+      if (env === 'test' && hasFactories) {
+        server.loadFactories(modulesMap['factories']);
+      } else if (hasDefaultScenario && hasFactories) {
```
Comment (samselikoff) [inline app/initializers/ember-cli-mirage.js]: I think we also need a `env !== 'test'` here, since we don't want these running during testing

**Comment by samselikoff** [comment]
module loading looks great, nice to get that cleaned up!

I'm not sure how to end to end test this either, since it's specifically not in `test` environment. I think what we should eventually do (not necessarily now) is create another object that understands the initializer logic, so we can unit test that.

For now if you make the one change I think we can ship this. I really appreciate your help!

**Comment by g-cassie** [comment]
Right now the behaviour for tests is to use factories if they exist and if not then to use fixtures.  Correct?

If we are supporting fixtures in tests shouldn't we support scenarios as well?  Or alternatively should we disable both fixtures and scenarios for tests?

**Comment by samselikoff** [comment]
Originally I conceived of scenarios as a development-only concept. You could boot the app in a default scenario, or a "new-user" scenario, or "pro-user" scenario, etc. Perhaps we'll use query params or a chrome extension to support loading different scenarios one day.

But I don't think scenarios makes sense for tests. Test themselves already are different scenarios, and I want to push for tests to set up their own state via factories. Not only is `/scenarios/default.js` far away from your tests (and thus make your tests less intention-revealing), I want to be able to set up a development scenario without it affecting all my tests. If we just loaded the scenario during tests, the tests wouldn't be atomic anymore and they'd all have to start from whatever data is in the scenario.

What do you think?

**Comment by g-cassie** [comment]
Personally I think factories is best practice in tests so you shouldn't use fixtures or scenarios.  I see scenarios and fixtures as being substantially identical though so would just question why one works for tests but the other doesn't.  Do you support multiple fixture suites right now?

Let me know your decision one way or the other and I'll make the change.

**Comment by samselikoff** [comment]
No we don't support multiple fixture suites, and I agree factories are best practice in tests. The whole point of scenarios is so people can use their factories to seed their database during development. I just want to keep those scenarios insulated from the test environment.

**Comment by g-cassie** [comment]
Ok, change made.

**Comment by samselikoff** [comment]
sweet, thanks!

**Comment by g-cassie** [comment]
Where is the code for the docs website?

**Comment by samselikoff** [comment]
On the `gh-pages` branch

**Pull request merged**